pkg/controller/utils/elasticsearch.go

```diff
@@ -588,20 +596,21 @@ func getESRoots(ctx context.Context, client client.Client) (*x509.CertPool, erro
 	return roots, nil
 }
 
-func extractPolicyDetails(policy map[string]interface{}) (string, string, string, error) {
+func extractPolicyDetails(policy map[string]interface{}) (string, string, string, bool, error) {
 	jsonPolicy, err := json.Marshal(policy)
 	if err != nil {
-		return "", "", "", err
+		return "", "", "", true, err
 	}
 	existingPolicy := Policy{}
 	if err = json.Unmarshal(jsonPolicy, &existingPolicy); err != nil {
-		return "", "", "", err
+		return "", "", "", true, err
 	}
 
 	currentMaxAge := existingPolicy.Phases.Hot.Actions.Rollover.MaxAge
 	currentMaxSize := existingPolicy.Phases.Hot.Actions.Rollover.MaxSize
 	currentMinAge := existingPolicy.Phases.Delete.MinAge
-	return currentMaxAge, currentMaxSize, currentMinAge, nil
+	readOnlyAfterRollover := existingPolicy.Phases.Warm.Actions.Readonly != nil
+	return currentMaxAge, currentMaxSize, currentMinAge, readOnlyAfterRollover, nil
```

Comment:
Do you know why we don't just do a rollover if anything has changed, i.e. do some sort of deep equal?

Reply:
That was my thought too when I first saw this, seems like it would be a lot simpler and better long term. But a bigger change could expose me to more unforeseen issues and wanted to keep the change straightforward.

The main issue I could foresee would be sensitivity to json formatting or ordering mismatch could cause the comparison to "fail" and the policy to be continuously updated. This could be avoided with proper testing. 

Would be happy to clean that up further if you think that's a good change?